spec/unit/provider/zfs/zfs_spec.rb

```diff
@@ -103,4 +103,23 @@
       end
     end
   end
+  describe "zoned" do
+    container = case Facter.value(:operatingsystem)
+      when "FreeBSD"
+        :jailed
+      else
+        :zoned
+    end
+    it "should get zoned" do
+      provider.expects(:zfs).with(:get, '-H', '-o', 'value', container, name).returns("value\n")
+
+      expect(provider.send("zoned")).to eq('value')
+    end
+
+    it "should set zoned=value" do
+      provider.expects(:zfs).with(:set, "#{container}=value", name)
+
+      provider.send("zoned=", "value")
+    end
+  end
```

Comment:
I think for these tests, what we want is to stub out the operating system and confirm that the zoned methods operate on the expected containers. I can give it a quick revision for you.